test/common/user_rhmi_developer_permissions.go

```diff
@@ -0,0 +1,98 @@
+package common
+
+import (
+	goctx "context"
+	"fmt"
+	"github.com/google/go-querystring/query"
+	"github.com/integr8ly/integreatly-operator/test/resources"
+	v1 "github.com/openshift/api/route/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"testing"
+)
+
+const (
+	expectedRhmiDeveloperNamespace    = "redhat-rhmi-fuse"
+	expectedRhmiDeveloperProjectCount = 1
+	expectedFusePodCount              = 6
+)
+
+// struct used to create query string for fuse logs endpoint
+type LogOptions struct {
+	Container string `url:"container"`
+	Follow    string `url:"follow"`
+	TailLines string `url:"tailLines"`
+}
+
+func TestRHMIDeveloperUserPermissions(t *testing.T, ctx *TestingContext) {
+	// get console master url
+	rhmi, err := GetRHMI(ctx)
+	if err != nil {
+		t.Fatalf("error getting RHMI CR: %v", err)
+	}
+	masterURL := rhmi.Spec.MasterURL
+
+	// get oauth route
+	oauthRoute := &v1.Route{}
+	if err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: resources.OpenshiftOAuthRouteName, Namespace: resources.OpenshiftAuthenticationNamespace}, oauthRoute); err != nil {
+		t.Fatal("error getting Openshift Oauth Route: ", err)
+	}
+
+	// get rhmi developer user tokens
+	rhmiDevUserToken, err := resources.DoAuthOpenshiftUser(oauthRoute.Spec.Host, masterURL, resources.DefaultIDP, "test-user01", "Password1")
+	if err != nil {
+		t.Fatalf("error occured trying to get token : %v", err)
+	}
+
+	// get projects for rhmi developer
+	rhmiDevfoundProjects, err := resources.DoOpenshiftGetProjects(masterURL, rhmiDevUserToken)
+	if err != nil {
+		t.Fatalf("error occured while getting user projects : %v", err)
+	}
+
+	// check if projects are as expected for rhmi developer
+	for projectCount, p := range rhmiDevfoundProjects.Items {
+		if projectCount >= expectedRhmiDeveloperProjectCount {
+			t.Fatalf("test failed - found rhmi developer project count : %s expected rhmi-developer project count : %s", projectCount, expectedRhmiDeveloperProjectCount)
+		}
+		if p.Name != expectedRhmiDeveloperNamespace {
+			t.Fatalf("test failed - found rhmi developer project: %s expected rhmi developer project : %s", p.Name, expectedRhmiDeveloperNamespace)
+		}
+	}
```

Comment:
Could this be simplified - just two asserts (no `for` loop):

- projectCount is as expected
- name is as expected